allennlp/training/gradient_descent_trainer.py

```diff
@@ -949,7 +954,9 @@ def _try_train(self) -> Tuple[Dict[str, Any], int]:
             epoch = self._num_epochs - 1
 
         # Load the best model state before returning
-        if self._best_model_filename is None or self._metric_tracker.is_best_so_far():
+        if self._should_validate_this_epoch and (
+            self._best_model_filename is None or self._metric_tracker.is_best_so_far()
+        ):
```

Comment:
I most unsure about this change

Reply:
Hmm I think we can omit the check here, i.e. leave it as it was.